### PR TITLE
fix transaction serialization

### PR DESCRIPTION
**Pull request by stonelgh**

### CODE REVIEW

File: transaction.go [hunk 1]
```diff
@@ -120,6 +120,14 @@ func NewTransaction(instructions []Instruction, blockHash PublicKey, opts ...Tra
 		finalAccounts[itr] = uniqAccount
 		itr++
 	}
+	if feePayerIndex < 0 {
+		// fee payer is not part of accounts we want to add it
+		finalAccounts[0] = &AccountMeta{
```
Comment (maoueh) [inline transaction.go:125]: Are we sure `finalAccounts` is always at least 1 element big?

File: account.go [hunk 2]
```diff
@@ -39,14 +39,11 @@ type AccountMeta struct {
 }
 
 func (a *AccountMeta) less(act *AccountMeta) bool {
-	if a.IsSigner && !act.IsSigner {
-		return true
-	} else if !a.IsSigner && act.IsSigner {
-		return false
+	if a.IsSigner != act.IsSigner {
+		return a.IsSigner
```
Comment (maoueh) [inline account.go:43]: | self (`a.IsIsigner`) | other (`act.IsSigner`) | Less |
|-|-|-|
|true|false|true|
|false|true|false|
|true|true|`<continue>`|
|false|false|`<continue>`|

Is there any details about the exact rules somewhere. Shouldn't the first two row be inverted?

Comment (stonelgh) [reply account.go:43]: I cannot find any documents specifying the rules. But I think solana-web3.js should be a good reference: [Transaction.compileMessage](https://github.com/solana-labs/solana-web3.js/blob/f34d5de/src/transaction.ts#L205)

Comment (maoueh) [reply account.go:43]: I agree, I will write the full tests later on.

**Pull request merged**